pkg/kv/kvserver/store.go

```diff
@@ -1861,10 +1861,12 @@ func (s *Store) Start(ctx context.Context, stopper *stop.Stopper) error {
 			// Uninitialized Replicas are not currently instantiated at store start.
 			continue
 		}
-		// TODO(sep-raft-log): construct the loaded Replica state directly in
-		// LoadAndReconcileReplicas, which loads and checks most of it already, then
-		// feed it to Replica creation, to avoid double-loading.
-		rep, err := loadInitializedReplica(ctx, s, repl.Desc, repl.ReplicaID)
+		// TODO(pavelkalinnikov): integrate into kvstorage.LoadAndReconcileReplicas.
+		state, err := repl.Load(ctx, s.Engine())
```

Comment:
You're not calling `state.Check()` here, which goes back to my earlier point (can't find the discussion now), should we bake the checks into `Load` so that they can't be skipped?

Reply:
I see that `Check` is called inside of `initRaftMuLockedReplicaMuLocked` so it is hit, but still, it's pretty unclear when this method is "supposed" to be called. Calling it automatically at state creation seems like the one reasonable place imo, what's the point of returning an invalid state in the first place?

Reply:
Yeah, sounds right, I'll see if `Load` looks better with the embedded check.

My rationale for separating them was related to the fact that we can load a state that doesn't pass the checks, fix it up (e.g. with migrations), and then `Check` before/during passing it to `Replica` constructor. But lemme do a simple thing first, and then revisit if needed.

Reply:
Done.